balance_service/enforcement/usage_enforcement.py

```diff
@@ -336,9 +348,24 @@ def stop_charging(self, data):
                     charge.end_time = now
                     charge.save()
 
-    def __get_billrate(self, resource, resource_type=None):
-        su_factor = resource.get("su_factor")
+    def __get_billrate(self, resource, reservation, resource_type=None):
+        # Do not charge for floating IPs
+        if resource_type == "virtual:floatingip":
+            return 0
+
+        # SU factor is configured per flavor in portal
+        # Note: this doesn't let us customize su_factor per project/user yet
+        if resource_type == "flavor:instance":
+            flavor_id = _get_reservation_flavor_id(reservation)
+            v = ConfigVariable.get_value(
+                "su_factor",
+                flavor_id=flavor_id,
+            )
+            if v is not None:
+                return float(v)
```

Comment:
what happens in this whole process is we don't have a value for the flavor in question (ie v is None)? what defaults does this revert back to?